packages/components/src/context-actions/ContextActions.tsx

```diff
@@ -87,32 +131,41 @@ class ContextActions extends Component {
     return { globalActions, keyboardActions };
   }
 
-  componentDidMount() {
-    if (this.container.parentNode) {
-      this.container.parentNode.addEventListener(
+  componentDidMount(): void {
+    if (this.container.current?.parentElement) {
+      this.container.current.parentElement.addEventListener(
         'contextmenu',
         this.handleContextMenu
       );
-      this.container.parentNode.addEventListener('keydown', this.handleKeyDown);
+      this.container.current.parentElement.addEventListener(
+        'keydown',
+        this.handleKeyDown
+      );
     }
   }
 
-  componentWillUnmount() {
-    if (this.container.parentNode) {
-      this.container.parentNode.removeEventListener(
+  componentWillUnmount(): void {
+    if (this.container.current?.parentElement) {
+      this.container.current.parentElement.removeEventListener(
         'contextmenu',
         this.handleContextMenu
       );
-      this.container.parentNode.removeEventListener(
+      this.container.current.parentElement.removeEventListener(
         'keydown',
         this.handleKeyDown
       );
     }
   }
 
-  handleContextMenu(e) {
-    if (!e.contextActions) {
-      e.contextActions = [];
+  container: React.RefObject<HTMLDivElement>;
```

Comment:
I think this should be put before the constructor

Reply:
I do too. This was where the current eslint config told me to put it. I've adjusted the config to place instance variables before the constructor, but after static members.

Reply:
Adjusting the config throws other things off w/ linting. We use `getCachedThing = memoize(...)` in places which is an instance variable. Might have to leave the eslint config and make these an exception to it?

Reply:
Yea I'm alright with leaving it... only other thing I can think of is moving the memoizes as well (need to think about that), or defining an interface first that it implements, eg.:

```
interface ContextActionsInterface {
  container: React.RefObject<HTMLDivElement>;
}

class ContextActions extends Component<...> implements ContextActionsInterface { ... }
```

Not sure that will actually work

Reply:
I'll make another ticket for adjusting the linter config so we can move these w/o needing rule overrides